File: MicroTube.Client.Angular/src/app/data/DTO/CommentSearchResultDTO.ts
import { CommentDto, CommentRawDto } from "./CommentDto";
export class CommentSearchResultDto
{
  comments: CommentDto[];
  meta: string | null;
  constructor(raw: CommentSearchResultRawDto)
  {
    this.comments = raw.comments.map(_ => new CommentDto(_));
    this.meta = raw.meta;
  }
}
export interface CommentSearchResultRawDto
{
  comments: CommentRawDto[];
  meta: string|null;
}

File: MicroTube/Controllers/Comments/DTO/CommentSearchResultDTO.cs
namespace MicroTube.Controllers.Comments.DTO
{
	public class CommentSearchResultDTO
	{

		public IEnumerable<CommentDTO> Comments { get; set; }
		public string? Meta { get; set; }
		public CommentSearchResultDTO(IEnumerable<CommentDTO> comments, string? meta)
		{
			Comments = comments;
			Meta = meta;
		}
	}
}
